pkg/guacrest/helpers/sbom.go

```diff
@@ -0,0 +1,154 @@
+//
+// Copyright 2024 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helpers
+
+import (
+	"context"
+	"fmt"
+	model "github.com/guacsec/guac/pkg/assembler/clients/generated"
+	"go.uber.org/zap"
+	"strings"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/Masterminds/semver"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+func LatestSBOMForAGivenId(ctx context.Context, client graphql.Client, id string) (*model.AllHasSBOMTree, error) {
```

Comment:
Can you write tests for this? See the tests in `pkg/guacrest/server/retrieveDependencies_test.go` as a reference.